### PR TITLE
Refactor start-build to use options style

### PR DESCRIPTION
**Pull request by smarterclayton**
Allows reuse of the command from other commands

### CODE REVIEW

**Comment by smarterclayton** [comment]
[test]

**Comment by smarterclayton** [comment]
@csrwng or @mfojtik

File: pkg/cmd/cli/cmd/startbuild.go [hunk 1]
```diff
@@ -84,62 +82,104 @@ func NewCmdStartBuild(fullName string, f *clientcmd.Factory, in io.Reader, out i
...
-		repo := git.NewRepository()
-		return RunStartBuildWebHook(f, out, webhook, path, postReceivePath, repo)
+		return nil
+
 	case len(args) != 1 && len(buildName) == 0:
 		return kcmdutil.UsageError(cmd, "Must pass a name of a build config or specify build name with '--from-build' flag")
 	}
 
+	if len(fromFile) > 0 || len(fromDir) > 0 || len(fromRepo) > 0 {
+		o.AsBinary = true
```
Comment (mfojtik) [inline pkg/cmd/cli/cmd/startbuild.go]:
```go
o.AsBinary = len(fromFile) > 0 || len(fromDir) > 0 || len(fromRepo) > 0
```

File: pkg/cmd/cli/cmd/startbuild.go [hunk 2]
```diff
@@ -155,86 +195,99 @@ func RunStartBuild(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra
...
 	if len(buildLogLevel) > 0 {
 		env = append(env, kapi.EnvVar{Name: "BUILD_LOGLEVEL", Value: buildLogLevel})
 	}
+	o.EnvVar = env
 
+	return nil
+}
+
+// RunStartBuild contains all the necessary functionality for the OpenShift cli start-build command
+func (o *StartBuildOptions) Run() error {
```
Comment (mfojtik) [inline pkg/cmd/cli/cmd/startbuild.go]: fix godoc.

**Comment by mfojtik** [comment]
@smarterclayton two nits, LGTM

**Comment by smarterclayton** [comment]
Comments addressed and rebased.

**Comment by mfojtik** [comment]
[merge]

**Pull request merged**